socorro/processor/processor_app.py

```diff
@@ -349,49 +241,42 @@ def process_crash(self, crash_id, ruleset_name, tmpdir):
 
         self.logger.info("completed %s", crash_id)
 
-    def _setup_source_and_destination(self):
+    def _set_up_source_and_destination(self):
         """Instantiate classes necessary for processing."""
-        self.queue = self.config.queue.crashqueue_class(
-            self.config.queue,
-            namespace=self.app_instance_name,
-        )
-        self.source = self.config.source.crashstorage_class(
-            self.config.source,
-            namespace=self.app_name,
-        )
-        self.destination = self.config.destination.crashstorage_class(
-            self.config.destination,
-            namespace=self.app_name,
-        )
-
-        self.processor = self.config.processor.processor_class(
-            config=self.config.processor, host_id=self.app_instance_name
-        )
-
-        self.temporary_path = self.config.processor.temporary_path
+        self.queue = build_instance_from_settings(settings.QUEUE)
+        self.source = build_instance_from_settings(settings.CRASH_SOURCE)
+        destinations = []
+        for key in settings.CRASH_DESTINATIONS_ORDER:
+            dest_obj = build_instance_from_settings(settings.CRASH_DESTINATIONS[key])
+            setattr(dest_obj, "crash_destination_name", key)
+            destinations.append(dest_obj)
+        self.destinations = destinations
+
+        self.pipeline = build_instance_from_settings(settings.PROCESSOR["pipeline"])
+
+        self.temporary_path = settings.PROCESSOR["temporary_path"]
         os.makedirs(self.temporary_path, exist_ok=True)
 
-        if self.config.companion_process.companion_class:
```

Comment:
I took out all the companion process stuff. There are a few problems with the companion process. The first is that it's kicked off as another thread in the processor process. The second is that it's unmanaged--if the thread dies, the processor just blithely continues on. The third is that it emits no metrics or logging, so we have no idea what it's doing. Also, since the stackwalker populates the LRU cache and it's not emitting metrics or logging either, we have no idea how effective the LRU cache is.

Anyhow, so I removed all the companion process stuff and wrote a new disk cache manager that runs as a separate process, emits metrics and logging, and when it dies, it causes the entire processor to die.